mx.py

```diff
@@ -2176,9 +2176,14 @@ def _pop_list(attrs, name, context):
     def _pop_os_arch(attrs, context):
         os_arch = attrs.pop('os_arch', None)
         if os_arch:
-            os_attrs = os_arch.pop(get_os(), None)
-            if not os_attrs:
-                os_attrs = os_arch.pop('<others>', None)
+            os_key = None
+            if get_os_variant():
+                os_key = get_os() + '-' + get_os_variant()
+            if os_key is None or not os_key in os_arch:
```

Comment:
Sorry for not noticing this in my previous pass: some linters like pep8 will complain about `not x in y` asking to write it as `x not in y` (same below).

After that change i think this is ready for integration (we can also do that style change during integration on our side).